sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -19,17 +19,39 @@ package org.apache.spark.sql.execution
 
 import java.util
 
-import org.apache.spark.rdd.RDD
+import org.apache.spark.annotation.DeveloperApi
 import org.apache.spark.sql.catalyst.expressions._
-import org.apache.spark.sql.catalyst.plans.physical.{AllTuples, ClusteredDistribution, Distribution, Partitioning}
+import org.apache.spark.sql.catalyst.plans.physical._
+import org.apache.spark.sql.types.IntegerType
+import org.apache.spark.rdd.RDD
 import org.apache.spark.util.collection.CompactBuffer
+import scala.collection.mutable
 
 /**
  * :: DeveloperApi ::
- * For every row, evaluates `windowExpression` containing Window Functions and attaches
- * the results with other regular expressions (presented by `projectList`).
- * Evert operator handles a single Window Specification, `windowSpec`.
+ * This class calculates and outputs (windowed) aggregates over the rows in a single sorted group.
+ * The aggregates are calculated for each row in the group. An aggregate can take a few forms:
+ * - Global: The aggregate is calculated for the entire group. Every row has the same value.
+ * - Rows: The aggregate is calculated based on a subset of the window, and is unique for each
+ * row and depends on the position of the given row within the window. The group must be sorted
+ * for this to produce sensible output. Examples are moving averages, running sums and row
+ * numbers.
+ * - Range: The aggregate is calculated based on a subset of the window, and is unique for each
+ * value of the order by clause and depends on its ordering. The group must be sorted for this to
+ * produce sensible output.
+ * - Shifted: The aggregate is a displaced value relative to the position of the given row.
+ * Examples are Lead and Lag.
```

Comment:
I agree. There are still a few other documentation inconsistencies, and I'll try to fix those as well.